src/gflags.h.in

```diff
@@ -429,11 +429,57 @@ extern GFLAGS_DLL_DECL void ShutDownCommandLineFlags();
 // people can't DECLARE_int32 something that they DEFINE_bool'd
 // elsewhere.
 
+enum FlagValueType {
+  FVT_BOOL = 0,
+  FVT_INT32 = 1,
+  FVT_UINT32 = 2,
+  FVT_INT64 = 3,
+  FVT_UINT64 = 4,
+  FVT_DOUBLE = 5,
+  FVT_STRING = 6,
+  FVT_MAX_INDEX = 6
+};
+
+template <typename T, const FlagValueType type>
+class FlagTraitsBase {
+ public:
+  typedef T ValueType;
+
+  static FlagValueType Type() { return type; }
+};
+
+template <typename T>
+class FlagTraits;
+
+template <>
+class FlagTraits<bool> : public FlagTraitsBase<bool, FVT_BOOL> {};
```

Comment:
I would like these not to show up in the `gflags.h` file. You can do this by following my advice above and only declare a template class (i.e., `FlagRegisterer`) in `gflags.h` and explicitly instantiated it for the needed types in `gflags.cc`. You only need those `FlagTraits` in the `FlagRegisterer` constructor definition.